subsys/bluetooth/host/iso.c

```diff
@@ -765,13 +765,39 @@ int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf)
 
 	iso_conn = chan->iso;
 
-	hdr = net_buf_push(buf, sizeof(*hdr));
-	hdr->sn = sys_cpu_to_le16(iso_conn->iso.seq_num);
-	hdr->slen = sys_cpu_to_le16(bt_iso_pkt_len_pack(net_buf_frags_len(buf)
-							- sizeof(*hdr),
-							BT_ISO_DATA_VALID));
+	/* Once the stored seq_num reaches the maximum value sendable to the
+	 * controller (BT_ISO_MAX_SEQ_NUM) we will allow the application to wrap
+	 * it. This ensures that up to 2^32-1 SDUs can be sent without wrapping
+	 * the sequence number which should provide amble room to handle
+	 * applications that does not send an SDU every interval.
+	 */
+	if (sn <= iso_conn->iso.seq_num &&
+	    iso_conn->iso.seq_num < BT_ISO_MAX_SEQ_NUM) {
+		BT_DBG("Invalid sn %u - Shall be > than %u",
+		       sn, iso_conn->iso.seq_num);
+		return -EINVAL;
+	}
+
+	iso_conn->iso.seq_num = sn;
```

Comment:
Nice, so even when the counter wraps after a minimum of 248 days, it will still work. Great for sensor data streaming, that is intended to run for years.

Reply:
Yes. Of course this also means that an application can do

```c
sn = 0x00001;
iso_send(sn); /* controller gets 0x0001 */
sn = 0x10001;
iso_send(sn); /* controller gets 0x0001 */
sn = 0x20001;
iso_send(sn); /* controller gets 0x0001 */
```
So it is still possible to abuse, but without any core spec specified value wrapping protocol this is unavoidable.